pkg/descheduler/descheduler.go

```diff
@@ -112,8 +112,9 @@ func (d *descheduler) runDeschedulerLoop(ctx context.Context, nodes []*v1.Node)
 	var span trace.Span
 	ctx, span = tracing.Tracer().Start(ctx, "runDeschedulerLoop")
 	defer span.End()
-	loopStartDuration := time.Now()
-	defer metrics.DeschedulerLoopDuration.With(map[string]string{}).Observe(time.Since(loopStartDuration).Seconds())
+	defer func(loopStartDuration time.Time) {
+		metrics.DeschedulerLoopDuration.With(map[string]string{}).Observe(time.Since(loopStartDuration).Seconds())
+	}(time.Now())
```

Comment:
Should not this be more like:
```go
loopStartDuration := time.Now()
defer func(loopStartDuration time.Time) {
		metrics.DeschedulerLoopDuration.With(map[string]string{}).Observe(time.Since(loopStartDuration).Seconds())
	}(loopStartDuration)
```
?

Reply:
I think both of these changes work. 

It's straightforward to use `defer func(...) {...} (time.Now())` because that's the only place the `loopStartDuration` variable is used, and it doesn't seem like it needs an extra variable defined.

Reply:
`time.Now()` will have different value for `loopStartDuration := time.Now()` and when `defer` is invoked.  Unless I incorrectly understand how the defer statement invokes the function/code. My assumption is anything after `defer` gets invoke right before a function/method exits.

Reply:
https://go.dev/tour/flowcontrol/12 says:
```
The deferred call's arguments are evaluated immediately, but the function call is not executed until the surrounding function returns.
```
Interesting.